### PR TITLE
Added code for Apigee Environments

### PR DESCRIPTION
**Pull request by priya-saxena21**
Added code for Apigee Environments

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_apigee_environment
```

### CODE REVIEW

File: mmv1/products/apigee/api.yaml [hunk 1]
```diff
@@ -206,4 +206,60 @@ objects:
...
+          - True
+          - False
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    description: |
+      An `Environment` in Apigee.
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'orgId'
```
Comment (melinath) [inline mmv1/products/apigee/api.yaml:236]: It looks like the API docs specify this as `parent` - why does `orgId` work? https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.environments/create

**Comment by melinath** [comment]
First build failed: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/175182

**Review by melinath — APPROVED** [review]
This LGTM from a code perspective, assuming the tests pass; it looks like there are some other fields that won't be supported yet, but they can always be added in separate PRs later.

**Comment by melinath** [comment]
Build 2: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/175186
Build 3: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/175188

**Comment by melinath** [comment]
Build succeeded: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/175293

The failing test is unrelated.

File: mmv1/products/apigee/api.yaml (hunk 1 above)
Comment (drebes) [reply mmv1/products/apigee/api.yaml:236]: parent is just a URL parameter, so we can call it however we want. We actually have the two models in other resources. For example, the older [`google_access_context_manager_access_level`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/access_context_manager_access_level) used parent as is, while the newer [`google_kms_crypto_key`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/kms_crypto_key) uses the parent as `key_ring`. The second (with an explicit field name that matches what is expected) seems more readable.

**Pull request merged**